From 44ee32359319fa92f3b496bbdc392609de1805ad Mon Sep 17 00:00:00 2001
From: Michael Bestas <mkbestas@lineageos.org>
Date: Sun, 15 Apr 2018 19:41:56 +0300
Subject: [PATCH 3/5] LineageParts: Fix loading default colors for non-RGB
 lights

* Commit Iab1dcad048de838bb9015c62a4171b002141f09a
  moved default color overlay loading from resetColors to
  mDefaultLowColor/mDefaultMediumColor/mDefaultFullColor
  but those were only set for RGB lights, so we always defaulted
  to no color on non-RGB lights when resetColors was run.
* Load default colors for non-RGB lights as well and do some minor cleanup

Change-Id: I578a895fe23b65ee50436091c4f31907fba9961d
---
 .../BatteryLightSettings.java                 | 26 +++++++++----------
 1 file changed, 13 insertions(+), 13 deletions(-)

diff --git a/src/org/lineageos/lineageparts/notificationlight/BatteryLightSettings.java b/src/org/lineageos/lineageparts/notificationlight/BatteryLightSettings.java
index baf1bc3..0a27e44 100644
--- a/src/org/lineageos/lineageparts/notificationlight/BatteryLightSettings.java
+++ b/src/org/lineageos/lineageparts/notificationlight/BatteryLightSettings.java
@@ -97,6 +97,13 @@ public class BatteryLightSettings extends SettingsPreferenceFragment implements
         mBatteryBrightnessZenPref =
                 (BatteryBrightnessZenPreference) prefSet.findPreference(BRIGHTNESS_ZEN_PREFERENCE);
 
+        mDefaultLowColor = res.getInteger(
+                com.android.internal.R.integer.config_notificationsBatteryLowARGB);
+        mDefaultMediumColor = res.getInteger(
+                com.android.internal.R.integer.config_notificationsBatteryMediumARGB);
+        mDefaultFullColor = res.getInteger(
+                com.android.internal.R.integer.config_notificationsBatteryFullARGB);
+
         mBatteryBrightness = mBatteryBrightnessPref.getBrightnessSetting();
 
         if (!pulsatingLed || segmentedBatteryLed) {
@@ -106,13 +113,6 @@ public class BatteryLightSettings extends SettingsPreferenceFragment implements
         if (mMultiColorLed) {
             setHasOptionsMenu(true);
 
-            mDefaultLowColor = res.getInteger(
-                    com.android.internal.R.integer.config_notificationsBatteryLowARGB);
-            mDefaultMediumColor = res.getInteger(
-                    com.android.internal.R.integer.config_notificationsBatteryMediumARGB);
-            mDefaultFullColor = res.getInteger(
-                    com.android.internal.R.integer.config_notificationsBatteryFullARGB);
-
             // Low, Medium and full color preferences
             mLowColorPref = (ApplicationLightPreference) prefSet.findPreference(LOW_COLOR_PREF);
             mLowColorPref.setOnPreferenceChangeListener(this);
@@ -163,20 +163,20 @@ public class BatteryLightSettings extends SettingsPreferenceFragment implements
         Resources res = getResources();
 
         if (mLowColorPref != null) {
-            int lowColor = LineageSettings.System.getInt(resolver, LineageSettings.System.BATTERY_LIGHT_LOW_COLOR,
-                    res.getInteger(com.android.internal.R.integer.config_notificationsBatteryLowARGB));
+            int lowColor = LineageSettings.System.getInt(resolver,
+                    LineageSettings.System.BATTERY_LIGHT_LOW_COLOR, mDefaultLowColor);
             mLowColorPref.setAllValues(lowColor, 0, 0, false);
         }
 
         if (mMediumColorPref != null) {
-            int mediumColor = LineageSettings.System.getInt(resolver, LineageSettings.System.BATTERY_LIGHT_MEDIUM_COLOR,
-                    res.getInteger(com.android.internal.R.integer.config_notificationsBatteryMediumARGB));
+            int mediumColor = LineageSettings.System.getInt(resolver,
+                    LineageSettings.System.BATTERY_LIGHT_MEDIUM_COLOR, mDefaultMediumColor);
             mMediumColorPref.setAllValues(mediumColor, 0, 0, false);
         }
 
         if (mFullColorPref != null) {
-            int fullColor = LineageSettings.System.getInt(resolver, LineageSettings.System.BATTERY_LIGHT_FULL_COLOR,
-                    res.getInteger(com.android.internal.R.integer.config_notificationsBatteryFullARGB));
+            int fullColor = LineageSettings.System.getInt(resolver,
+                    LineageSettings.System.BATTERY_LIGHT_FULL_COLOR, mDefaultFullColor);
             mFullColorPref.setAllValues(fullColor, 0, 0, false);
             updateBrightnessPrefColor(fullColor);
         }
-- 
2.17.0

